wowchemy/assets/js/wowchemy-slides.js

```diff
@@ -0,0 +1,146 @@
+import * as params from "@params";
+
+import {fixMermaid} from './wowchemy-utils';
+
+// Enable core slide features.
+enabledPlugins = [
+  RevealMarkdown,
+  RevealHighlight,
+  RevealSearch,
+  RevealNotes,
+  RevealMath,
+  RevealZoom,
+];
+
+const isObject = function (o) {
+  return o === Object(o) && !isArray(o) && typeof o !== "function";
+};
+
+const isArray = function (a) {
+  return Array.isArray(a);
+};
+
+const toCamelCase = (s) => {
+  return s.replace(/([-_][a-z])/gi, ($1) => {
+    return $1.toUpperCase().replace("-", "").replace("_", "");
+  });
+};
+
+const keysToCamelCase = function (o) {
+  if (isObject(o)) {
+    const n = {};
+
+    Object.keys(o).forEach((k) => {
+      n[toCamelCase(k)] = keysToCamelCase(o[k]);
+    });
+
+    return n;
+  } else if (isArray(o)) {
+    return o.map((i) => {
+      return keysToCamelCase(i);
+    });
+  }
+
+  return o;
+};
+
+// reveal configurations can be included in front matter under slides.reveal
+var pluginOptions = {}
+if(typeof params.slides.reveal_options === "undefined"){
+  pluginOptions = params.slides.reveal_options;
+}
+
+
+pluginOptions = keysToCamel(pluginOptions);
+
+// Enable presentation menu by default.
+if (pluginOptions.menu_enabled === undefined) {
+  pluginOptions.menu_enabled = true;
+}
+
+// configure menu if enabled
+if (pluginOptions.menu_enabled) {
+  enabledPlugins.push(RevealMenu);
+
+  // make sure we have a menu configuration so we can set defaults
+  pluginOptions.menu = {}
+  if(typeof pluginOptions.menu === "undefined"){
```

Comment:
Why are you checking if `pluginOptions.menu` is undefined when you just set it to `pluginOptions.menu = {}` in the line above?

Reply:
should be removed because of the way we are now passing configuration to plugins .. there are also no defaults since chalkboard not included.